### ISSUE TITLE
A possible (cosmetic?) bug in TileExporter in splitRegionRequests

### ISSUE THREAD

**Issue by juliangilbey** [qupath/qupath#649]
In `splitRegionRequests`, the working coordinates are in downscaled pixels, and stored in `maxX` and `maxY`.  However, the two nested `for` loops over `x` and `y` have incorrect initialisers: for `y` it is `int y = request.getY()` instead of `int y = (int)(request.getMinY() / downsample)`.  It probably doesn't matter much, though, as presumably `request.getY()` and `request.getMinY()` both return 0 in normal situations.

**Comment by petebankhead** [comment]
I think you're right, fortunately I don't see any obvious code path that could call it with anything other than 0 for those values but I'll look a bit deeper before the next release.

**Comment by juliangilbey** [comment]
Another thing I've just spotted in this method: it uses both `request.getWidth()` to set `fullWidth` and `request.getMaxX()` later.  If the request has `request.getMinX() == 0`, as it appears it must do, then `request.getWidth()` and `request.getMaxX()` should return the same value.  I personally think that `request.getMaxX()` makes more semantic sense here; the way `fullWidth` is used is really to mean `maxX`.  (The same obviously goes for `maxY` and `fullHeight`.)

**Comment by juliangilbey** [comment]
You might like to look at my fork https://github.com/juliangilbey/qupath/tree/main+tileexporterregion where I've addressed all of the above points when writing a new class `TileExporterRegion` (`TileExporterRegion.java`) that is based on TileExporter but allows a sub-region to be specified.  You can therefore say:

```groovy
new TileExporterRegion(imageData)
    .downsample(downsample)   // Define export resolution
    // .requestedPixelSize(requestedPixelSize) // Alternative for defining export resolution
    .regionLimits(x, y, width, height) // Define region to be tiled
    .imageExtension('.tif')   // Define file extension for original pixels
                              // (often .tif, .jpg, '.png' or '.ome.tif')
    .tileSize(512)            // Define size of each tile, in pixels
    .annotatedTilesOnly(false) // If true, only export tiles if there is a
                              // (classified) annotation present
    .overlap(64)              // Define overlap, in pixel units at the
                              // export resolution
    .includePartialTiles(false) // don't include tiles that overlap the boundary
    .writeTiles(pathOutput)   // Write tiles to the specified directory
```
It can be used as a drop-in replacement for the current `TileExporter.java` file (obviously replacing every `TileExporterRegion` with `TileExporter`), as the default behaviour is to tile the entire image.

Note that the `regionLimits` method only specifies the x and y limits; it does not limit t or z.  That could be performed with separate methods, say `tLimits` and `zLimits`, if desired.

**Comment by petebankhead** [comment]
Fixed in https://github.com/qupath/qupath/pull/680 - will be included in v0.3.0.
